### PR TITLE
[tests-only] [full-ci] Refactor and fix dav-path-spaces test code

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
When running on reva, the personal space id has to be looked up by more tricky ways. That code is coming. But at the moment some test scenarios are failing for new and old dav path because code in TusContext calls `getPersonalSpaceIdForUser` even though the scenario is not actually using the "spaces dav path".

CI has been green in oCIS because the call to ऽgetPersonalSpaceIdForUser` does work on oCIS. The call is wasted, the space id is not actually used, but at least the code runs.

On `cs3org/reva` the space id is not found, because the `/graph/v1.0/me/drives` endpoint does not exist. And those scenarios fail when they should not fail.

https://drone.cernbox.cern.ch/cs3org/reva/5429/15/6
`Notice: Trying to get property 'value' of non-object in /drone/src/tmp/testrunner/tests/TestHelpers/WebDavHelper.php line 401`

This PR does:
1) refactor test code so that the "davPathVersion" magic numbers 1,2,3 are held in constants `DAV_VERSION_OLD` `DAV_VERSION_NEW` `DAV_VERSION_SPACES`. That should make code easier to read, rather than having 1,2,3 mentioned in multiple places.

2) modify `FeatureContext` `getPersonalSpaceIdForUser` so that it only actually tries to get the space id if the scenario is using `DAV_VERSION_SPACES`. If not, then it returns `null`. That will allow it to be called more flexibly.

## How Has This Been Tested?
CI

oCIS PR https://github.com/owncloud/ocis/pull/3105 passes against this branch, so the changes do not break that.

cs3org/reva PR https://github.com/cs3org/reva/pull/2497 passes against this branch, with just sensible line number changes in the expected-failures files.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/TestHelpers/WebDavHelper.php [hunk 1]
```diff
@@ -558,13 +561,18 @@ public static function getDavPath(
 		if ($type === "customgroups") {
 			return "remote.php/dav/";
 		}
-		if ($davPathVersionToUse === 3) {
+		if ($davPathVersionToUse === self::DAV_VERSION_SPACES) {
+			if (($spaceId === null) || (\strlen($spaceId) === 0)) {
+				throw new InvalidArgumentException(
+					__METHOD__ . " A spaceId must be passed when using DAV path version 3 (spaces)"
+				);
+			}
```
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:569]: :100:

File: tests/acceptance/features/bootstrap/ChecksumContext.php [hunk 2]
```diff
@@ -163,11 +163,6 @@ public function userRequestsTheChecksumOfViaPropfind(string $user, string $path)
 				<oc:checksums />
 			  </d:prop>
 			</d:propfind>';
-		$url = $this->featureContext->getBaseUrl() . '/' .
-			WebDavHelper::getDavPath(
-				$user,
-				$this->featureContext->getDavPathVersion()
-			) . $path;
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/ChecksumContext.php:170]: `$url` is never used afterward. :+1:

**Review by kiranparajuli589 — APPROVED** [review]
looks good

File: tests/TestHelpers/WebDavHelper.php (hunk 1 above)
Comment (phil-davis) [reply tests/TestHelpers/WebDavHelper.php:569]: This helped me find bugs!

**Pull request merged**